src/Explorer/Tabs/DocumentsTabContent.tsx

```diff
@@ -0,0 +1,811 @@
+        query,
+        undefined
+      );
+      if (queryDocumentsData) {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const nextDocumentIds = queryDocumentsData.documents.map((rawDocument: any) => {
+          const partitionKeyValue = rawDocument._partitionKeyValue;
+          return new DocumentId(this.props as DocumentsTab, rawDocument, partitionKeyValue);
+        });
+        this.setState({ documentIds: nextDocumentIds });
+      }
+      if (this.props.onLoadStartKey !== undefined) {
+        TelemetryProcessor.traceSuccess(
+          Action.Tab,
+          {
+            databaseName: this.props.collection.databaseId,
+            collectionName: this.props.collection.id(),
+            dataExplorerArea: Areas.Tab,
+            tabTitle: this.props.tabTitle(),
+          },
+          this.props.onLoadStartKey
+        );
+      }
+      this.props.isExecuting(false);
+    } catch (error) {
+      if (this.props.onLoadStartKey !== undefined) {
+        TelemetryProcessor.traceFailure(
+          Action.Tab,
+          {
+            databaseName: this.props.collection.databaseId,
+            collectionName: this.props.collection.id(),
+            dataExplorerArea: Areas.Tab,
+            tabTitle: this.props.tabTitle(),
+            error: getErrorMessage(error),
+            errorStack: getErrorStack(error),
+          },
+          this.props.onLoadStartKey
+        );
+      }
+      this.props.isExecuting(false);
+    }
+  };
+
+  private handleRow = (row: DocumentId | Resource): void => {
+    if (this.state.isEditorContentEdited) {
+      const isChangesConfirmed = window.confirm("Your unsaved changes will be lost.");
+      if (isChangesConfirmed) {
+        this.handleRowContent(row);
+        this.setState({ isEditorContentEdited: false });
+        return;
+      }
+    } else {
+      this.handleRowContent(row);
+    }
+  };
+
+  private handleRowContent = (row: DocumentId | Resource): void => {
+    const formattedDocumentContent =
+      userContext.apiType === "Mongo"
+        ? formatDocumentContent(row as DocumentId)
+        : formatSqlDocumentContent(row as Resource);
+    this.updateTabButtonVisibility();
+
+    userContext.apiType === "Mongo"
+      ? this.updateContent(row as DocumentId, formattedDocumentContent)
+      : this.updateSqlContent(row as Resource, formattedDocumentContent);
+  };
+
+  private updateContent = (row: DocumentId, formattedDocumentContent: string): void => {
+    this.setState(
+      {
+        documentContent: formattedDocumentContent,
+        isEditorVisible: true,
+        editorKey: row.rid,
+        selectedDocumentId: row,
+      },
+      () => {
+        this.updateTabButton();
+      }
+    );
+  };
+
+  private updateSqlContent = (row: Resource, formattedDocumentContent: string): void => {
+    this.setState(
+      {
+        documentContent: formattedDocumentContent,
+        isEditorVisible: true,
+        editorKey: row._rid,
```

Comment:
From what I'm seeing, `editorKey` is usually set at the same time as `documentContent`. It seems redundant with `documentContent`.
`documentContent` should be sufficient to pass to the editor to re-render (as `content`): if it changes, the editor will re-render.
Are we trying to force the editor to re-render with no content change? If so, can you why is it really necessary?

Reply:
Actually, Initially React editor was not re-rending on content change. But now working fine. Removed editorKey completely.